maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/util/DependencyScanner.java

```diff
@@ -111,19 +112,46 @@ private static void scanArtifact( File artifact, TestFilter<String, String> filt
         {
             for ( String groups : groupArtifactIds )
             {
+                // groupId:artifactId[:version[:type[:classifier]]]
                 String[] groupArtifact = groups.split( ":" );
-                if ( groupArtifact.length != 2 )
+                if ( groupArtifact.length < 2 || groupArtifact.length > 5 )
                 {
                     throw new IllegalArgumentException( "dependencyToScan argument should be in format"
-                        + " 'groupid:artifactid': " + groups );
+                        + " 'groupid:artifactid[:version[:type[:classifier]]]': " + groups );
                 }
-                if ( artifact.getGroupId().matches( groupArtifact[0] )
-                    && artifact.getArtifactId().matches( groupArtifact[1] ) )
+                if ( artifactMatchesGavtc( artifact, groupArtifact ) )
                 {
                     matches.add( artifact.getFile() );
                 }
             }
         }
         return matches;
     }
+    
+    private static boolean artifactMatchesGavtc( Artifact artifact, String[] gavtc )
+    {
+        boolean match = false;
+        if ( artifact.getGroupId().matches( gavtc[0] ) && artifact.getArtifactId().matches( gavtc[1] ) )
+        {
+            match = true;
+            // Check version
```

Comment:
I mean only these comments. For me it is quite obvious what you mean in code and thus the comment is useless. If you want to put a meaning on the IF statement, you can do something like this:
`if ( hasGroupAndArtifactId( groups ) )`
...
```private static boolean hasGroupAndArtifactId( String groups )
    return StringUtils.countMatches( groups, ':' ) >= 2;
```
The project knows several `StringUtils` but I mean the one from library `commons-lang3`.